util/dvsim/Modes.py

```diff
@@ -77,19 +77,60 @@ def merge_mode(self, mode):
         if not mode.name == self.name and not is_sub_mode:
             return False
 
-        # only merge the lists; if strs are different, then throw an error
-        attrs = self.__dict__.keys()
-        for attr in attrs:
-            # merge lists together
-            self_attr_val = getattr(self, attr)
-            mode_attr_val = getattr(mode, attr)
-
-            if type(self_attr_val) is list and type(mode_attr_val) is list:
+        # Merge attributes in self with attributes in mode arg, since they are
+        # the same mode but set in separate files, or a sub-mode.
+        for attr, self_attr_val in self.__dict__.items():
+            mode_attr_val = getattr(mode, attr, None)
+
+            # If sub-mode, skip the name fields - they could differ.
+            if is_sub_mode and attr in ['name', 'mname']:
+                continue
+
+            # If mode's value is None, then nothing to do here.
+            if mode_attr_val is None:
+                continue
+
+            # If self value is None, then replace with mode's value.
+            if self_attr_val is None:
+                setattr(self, attr, mode_attr_val)
+                continue
```

Comment:
This is new behaviour, I think. Is that intentional?

Reply:
Not really. This was happening in `check_conflict` method that is now removed.

Reply:
Oh, because in this case `self_attr_val` was equal to `default_val`. Sorry, I didn't spot that. The new code is much clearer! :+1: